pydantic/fields.py

```diff
@@ -292,7 +306,9 @@ class MyModel(pydantic.BaseModel):
                 new_field_info.frozen = final or field_info.frozen
                 metadata: list[Any] = []
                 for a in extra_args:
-                    if not isinstance(a, FieldInfo):
+                    if _typing_extra.is_deprecated_instance(a):
+                        new_field_info.deprecated = a.message
```

Comment:
Ah I see, so we could add further support for `deprecated` instance data here in the future, if desired.

Reply:
It's not much work: https://github.com/pydantic/pydantic/commit/cf06a7322d59aefc2e3ca607f0c5ee66d28f6757

but can be added after this PR